File: Exercicio4/TratamentoExcecoesC++/TratExcecoes.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <system_error>
#include <filesystem>
#include <vector>
#include <experimental/filesystem>

namespace fs = std::filesystem;

std::string lerConteudoArquivo(const std::string& nomeArquivo) {
    try {
        std::ifstream file(nomeArquivo, std::ios::binary);
        if (!file.is_open()) {
            return "ERRO: Arquivo não encontrado: " + nomeArquivo;
        }
        std::string conteudo((std::istreambuf_iterator<char>(file)), 
        std::istreambuf_iterator<char>());
        
        return conteudo;
        
    } catch (const std::system_error& e) {
        return "ERRO ao ler arquivo " + nomeArquivo + ": " + e.what();
    } catch (const std::exception& e) {
        return "ERRO inesperado ao ler arquivo " + nomeArquivo + ": " + e.what();
    }
}

File: Exercicio4/TratamentoExcecoesP/TratExcecoes.py
def ler_conteudo_arquivo(nome_arquivo):
    try:
        with open(nome_arquivo, 'r', encoding='utf-8') as arquivo:
            return arquivo.read()
            
    except FileNotFoundError:
        return f"ERRO: Arquivo não encontrado: {nome_arquivo}"
    except PermissionError:
        return f"ERRO: Permissão negada para ler o arquivo: {nome_arquivo}"
    except IOError as e:
        return f"ERRO ao ler arquivo {nome_arquivo}: {str(e)}"
    except Exception as e:
        return f"ERRO inesperado ao ler arquivo {nome_arquivo}: {str(e)}"